### PR TITLE
[prototype] Error Propagation Demo 

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Intro to Error Propagation with PennyLane**

**Summary:**
Highlight how a user could "hack" in some simple error tracking and propagation into PennyLane.

----
If you are writing a demonstration, please answer these questions to facilitate the marketing process.

GOALS — Why are we working on this now?

  *Eg. Promote a new PL feature or show a PL implementation of a recent paper.*
- Prototype / inspiration for new features 
- To address a need for researchers doing error propagation by hand with pennylane

AUDIENCE — Who is this for?

  *Eg. Chemistry researchers, PL educators, beginners in quantum computing.*
- Researchers, Industry experts 

KEYWORDS — What words should be included in the marketing post?*
- Error, Error Tracking, Resource Estimation

Which of the following types of documentation is most similar to your file? 
(more details [here](https://www.notion.so/xanaduai/Different-kinds-of-documentation-69200645fe59442991c71f9e7d8a77f8))
    
- [ ] Tutorial
- [ ] Demo
- [x] How-to

### CODE REVIEW

File: demonstrations/tutorial_error_prop.metadata.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }
+    ],
+    "seoDescription": "Learn how to propagate error through quantum circuits.",
+    "doi": "",
+    "canonicalURL": "/qml/demos/tutorial_error_prop",
+    "references": [],
+    "basedOnPapers": [],
+    "referencedByPapers": [],
+    "relatedContent": []
+}
```
Comment (Jaybsoni) [inline demonstrations/tutorial_error_prop.metadata.json:28]:
```suggestion
}

```

File: demonstrations/tutorial_error_prop.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+# In this demo, we showcased the :class:`~.pennylane.ResourcesOperation`, and the 
+# :class:`~.pennylane.error_prop.OperatorError` classes in PennyLane. We explained how to construct 
+# a custom resource operation and custom error type. We used this in a simple circuit to track and 
+# propagate the error through the circuit to the final measurement. We hope that you can use this 
+# tools in cutting edge research workflows to estimate error. 
+#
+##############################################################################
+# About the author
+# ----------------
+# .. include:: ../_static/authors/jay_soni.txt
```
Comment (Jaybsoni) [inline demonstrations/tutorial_error_prop.py:228]:
```suggestion
# .. include:: ../_static/authors/jay_soni.txt

```

**Comment by Jaybsoni** [comment]
[sc-50233]

File: demonstrations/tutorial_error_prop.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+            num_gates=len(self.decomposition()),
+            gate_types={"Trotter_w_Error": 1},
+            gate_sizes={1:1},
+            depth=len(self.decomposition()),
+            error=SpectralNorm_Error(op_error),
+        )
+
+
+###############################################################################
+# Problem:
```
Comment (trbromley) [inline demonstrations/tutorial_error_prop.py:113]:
```suggestion
# Problem
```

**Review by trbromley — COMMENTED** [review]
Thanks @Jaybsoni! This will be a nice basis for discussions.

Do you agree with this timeline:
- Use this PR to get feedback
- We make a PRD this quarter on how we want the functionality to look as a fully-fledged feature in PL
- Implement in Q1, and at the same time update this demo / make a new one using the new PL functionality

**Comment by Jaybsoni** [comment]
> * demo

Sounds good @trbromley 👍🏼

**Comment by Jaybsoni** [comment]
Closing this prototype as it has since been updated into a separate [PR](https://github.com/PennyLaneAI/qml/pull/1083)

**Pull request closed without merging**